rmgpy/solver/base.pyx

```diff
@@ -978,33 +991,33 @@ cdef class ReactionSystem(DASx):
                     logging.info('Surface now has {0} Species and {1} Reactions'.format(len(self.surfaceSpeciesIndices),len(self.surfaceReactionIndices)))
                     
             if filterReactions:
-                # Calculate thresholds for reactions
-                (unimolecularThresholdRateConstant,
-                 bimolecularThresholdRateConstant,
-                 trimolecularThresholdRateConstant) = self.get_threshold_rate_constants(modelSettings)
-                unimolecularThresholdVal = toleranceMoveToCore * charRate / unimolecularThresholdRateConstant
-                bimolecularThresholdVal = toleranceMoveToCore * charRate / bimolecularThresholdRateConstant
-                trimolecularThresholdVal = toleranceMoveToCore * charRate / trimolecularThresholdRateConstant
+                unimolecularThresholdVal = toleranceMoveToCore * charRate / kvals_uni
+                bimolecularThresholdVal = toleranceMoveToCore * charRate / kvals_bi
+                trimolecularThresholdVal = toleranceMoveToCore * charRate * 1e3 / kvals_bi
+
                 for i in xrange(numCoreSpecies):
-                    if not unimolecularThreshold[i]:
-                        # Check if core species concentration has gone above threshold for unimolecular reaction
-                        if coreSpeciesConcentrations[i] > unimolecularThresholdVal:
-                            unimolecularThreshold[i] = True
+                    for k in xrange(len(unimolecularThresholdVal)):
+                        if not unimolecularThreshold[i,k]:
+                            # Check if core species concentration has gone above threshold for unimolecular reaction
+                            if coreSpeciesConcentrations[i] > unimolecularThresholdVal[k]:
+                                unimolecularThreshold[i,k] = True
                 for i in xrange(numCoreSpecies):
                     for j in xrange(i, numCoreSpecies):
-                        if not bimolecularThreshold[i,j]:
-                            if coreSpeciesConcentrations[i]*coreSpeciesConcentrations[j] > bimolecularThresholdVal:
-                                bimolecularThreshold[i,j] = True
+                        for k in xrange(len(bimolecularThresholdVal)):
+                            if not bimolecularThreshold[i,j,k]:
+                                if coreSpeciesConcentrations[i]*coreSpeciesConcentrations[j] > bimolecularThresholdVal[k]:
+                                    bimolecularThreshold[i,j,k] = True
```

Comment:
Would this also be faster as a matrix operation (including uni and trimolecular)? You would end up filling the entire matrix/tensor though.